src/components/tile-select/tile-select.component.tsx

```diff
@@ -26,30 +24,81 @@ import {
 } from "./tile-select.style";
 import { filterStyledSystemMarginProps } from "../../style/utils";
 
-const marginPropTypes = filterStyledSystemMarginProps(
-  styledSystemPropTypes.space
-);
+const checkPropTypeIsNode = (prop: unknown): boolean =>
+  typeof prop !== "string";
 
-const checkPropTypeIsNode = (prop) => typeof prop !== "string" && { as: "div" };
+export interface CustomEvent {
```

Comment:
I think that we should have a meaningful name for this event - e.g.: `TileSelectChangeEvent`